src/components/inputs/select-input/select-input.js

```diff
@@ -157,18 +231,29 @@ export class SelectInput extends React.Component {
                   }
                 : undefined
             }
-            noOptionsMessage={
-              this.props.noOptionsMessage ||
-              (({ inputValue }) =>
-                inputValue === ''
-                  ? this.props.intl.formatMessage(
-                      messages.noOptionsMessageWithoutInputValue
-                    )
-                  : this.props.intl.formatMessage(
-                      messages.noOptionsMessageWithInputValue,
-                      { inputValue }
-                    ))
+            onChange={selectedOptions =>
+              // selectedOptions is either an array, or a single option
+              // depending on whether we're in multi-mode or not (isMulti)
+              this.props.onChange({
+                target: {
+                  name: this.props.name,
+                  // eslint-disable-next-line no-nested-ternary
```

Comment:
can this nested ternary be moved in a if statement or something else?

Reply:
It can become 

```js
                  value: do {
                    if (!selectedOptions) selectedOptions;
                    else if (this.props.isMulti)
                      selectedOptions.map(option => option.value);
                    else selectedOptions.value;
                  },
```

but I'm not sure this reads more clearly. I'd keep it. With prettier guaranteeing a good formatting of the nested ternary it isn't confusing to read (to me).

Reply:
personally I would have preferred the `do` expression here, but this is fine for me

Reply:
personally, I'd also have preferred the do expression